libs/state/docs/rx-effects.md

```diff
@@ -0,0 +1,182 @@
+# Resources
+
+**Example applications:**  
+A demo application is available on [GitHub](https://github.com/BioPhoton/rx-angular-state-rx-effects).
+
+# Motivation
+
+In certian cases we have some side effects present in components.
+As side effects by definition do not introduce state changes directly the management and maintainance of side effecs should not be tied to the `RxState` service.
+For this very reason all side effect management is organized in the `RxEffecfts` service.
+
+Side effects in general are always a potential leak in our architecture or memory it self if not cleaned up correctly.
+In Angular specifically as they should be tightly coupled to the component life cycle and get cleaned up at latest when the component got cleaned up.
+
+If we not only consider imperative but also reactively drive side effects a clean implementation ends up to be pretty wordy and boilerplatey.
+
+We have to get hold of Angular's `OnDestroy` method and either unsubscribe in that method, or use it as trigger in other places with e.g. the `takeUntil`operator.
+
+Using `takeUntil` to run the clean up logic.
+
+```ts
+@Component({
+  // ...
+})
+export class FooComponent implements OnDestroy {
+  // ⚠ Notice: The destroy hook must be reactive to use `takeUntil`
+  private readonly destroy$ = new Subject<void>();
+
+  constructor() {
+    effect$
+      .pipe(
+        takeUntil(this.destroy$)
+        // ⚠ Notice: Don't put any operator after takeUntil to avoid potential subscription leaks
+      )
+      .subscribe(doSideEffect);
+  }
+
+  ngOnDestroy(): void {
+    // ⚠ Notice: Never forget to cleanup the subscription
+    this.destroy$.next();
+  }
+}
+```
+
+Using the `subscription` to run the clean up logic.
+
+```ts
+@Component({
+  // ...
+})
+export class FooComponent implements OnDestroy {
+  // ⚠ Notice: The created subscriptin must be stored to `unsubscribe` later
+  private readonly subscription: Subscription;
+
+  constructor() {
+    // ⚠ Notice: Never forget to store the subscription
+    this.subscription = effect$.subscribe(doSideEffect);
+  }
+
+  ngOnDestroy(): void {
+    // ⚠ Notice: Never forget to cleanup the subscription
+    this.subscription.unsubscribe();
+  }
+}
+```
+
+In RxAngular we think the essential problem here is the call to `subscribe` itself. Same as `RxState`, `RxEffects` is a component only provider that is tied to the components life cycle.
+We can hand over `Observables` for the reactive side effects as well as a `Subscription` from other already started `Observables` the a method called `register`.
+To also provide an imperative way for developers to unsubscribe from the side effect `register` returns an "asyncId" similar to `setTimeout`.
+This can be used later on to call `unregister` and pass the async id retreived from a previous `register` call. This stops and cleans up the side effect when invoked.
```

Comment:
```suggestion
In RxAngular we think the essential problem here is the call to `subscribe` itself. All `Subscription`s need to get unsubscribed manually which most of the time produces heavy boilerplate or even memory leaks if ignored or did wrong. 
Like `RxState`, `RxEffects` is a local service provided by a component and thus tied to the components life cycle.
We can manage `Observables` as reactive triggers for side effects or manage `Subscription`s which internally hold side effects.
To also provide an imperative way for developers to unsubscribe from the side effect `register` returns an "asyncId" similar to `setTimeout`.
This can be used later on to call `unregister` and pass the async id retreived from a previous `register` call. This stops and cleans up the side effect when invoked.
```